### PR TITLE
Python3branch

### PR DESCRIPTION
**Pull request by wkbkhard**
Maybe this could be a new release for Plone 5.2 with Python3 - would be great, so we can get it with Buildout directly

### CODE REVIEW

File: plone/formwidget/masterselect/widget.py [hunk 1]
```diff
@@ -207,6 +208,7 @@ def __call__(self):
         self.request.response.setHeader(
             'Content-Type', 'application/json; charset=utf-8')
 
+        # import pdb; pdb.set_trace()
```
Comment (tisto) [inline plone/formwidget/masterselect/widget.py:211]: @wkbkhard accident? :)

**Comment by tisto** [comment]
@wkbkhard are you using your branch in production? PR looks good. Seems like the Travis config is somehow broken.

File: plone/formwidget/masterselect/widget.py (hunk 1 above)
Comment (wkbkhard) [reply plone/formwidget/masterselect/widget.py:211]: whoops - definitely accident. I will update it as soon as possible!

**Comment by tisto** [comment]
@wkbkhard I looked into this briefly and fixed the Travis setup:

https://travis-ci.org/github/collective/plone.formwidget.masterselect/jobs/696488974

Though, the Robot tests now fail with assertion errors. I just did this as a favor for a client and I can not put much effort into this. Though, if you would look into the failing tests and could fix them I could help to get a release out of the door.

**Comment by tisto** [comment]
Hmm. Master is broken since we merged this PR:

https://github.com/collective/plone.formwidget.masterselect/pull/16
https://travis-ci.org/github/collective/plone.formwidget.masterselect/builds/286453318

I have no idea if those Robot tests ever worked. It would be nice to fix them. Though, I am not sure if that should be a blocker for a release. This pkg was released before with Travis red...

**Comment by tisto** [comment]
@wkbkhard I ran the robot tests locally and they fail because the master-select widget functionality is not working, at least not as far as I understand how this is supposed to work.

**Comment by wkbkhard** [comment]
Could you be a bit more specific? What exactly does not work? The change of fields in the frontend? Something in the backend?

I'll take a look on the weekend then.

**Comment by tisto** [comment]
@wkbkhard you can run "bin/test" locally. If you have Firefox and geckodriver installed a Firefox instance will show up automatically. When I change one of the master select fields I would expect the other fields to change dependent on the master selection. This does not happen in any of the fields in the tests though. It could be that the fields in the tests are not properly set up and that it works in real life...

**Comment by wkbkhard** [comment]
I'll take a look and hopefully can update the tests, so they will work again.

**Comment by holdenhao** [comment]
Using python3branch-travis branch on Plone 5.2.2.  Master select features work on the demo form and also in my test content type.

**Comment by tisto** [comment]
@wkbkhard any chance you can have a look at the failing tests? It would be really nice if we could merge this and make a release.

**Review by tdesvenain — APPROVED** [review]

**Comment by tisto** [comment]
Anyone who is willing to look into the test failures so we can make a release?

**Comment by holdenhao** [comment]
The setuptools upgrade loop can be fixed by setting newest=false in buildout.cfg. For more details: https://github.com/buildout/buildout/issues/312#issuecomment-455557349

**Comment by laulaz** [comment]
Buildout has been fixed / upgraded to 5.2.2 on #19

Travis builds are now succeeding 👍 

Can we close this one ?

**Pull request merged**